### ISSUE TITLE
Difficulty reproducing DROW3 results

### ISSUE THREAD

**Issue by john-maidbot** [VisualComputingInstitute/2D_lidar_person_detection#9]
Hello,

I am trying to reproduce the results from the DROW3 paper: https://arxiv.org/pdf/1804.02463.pdf

I downloaded the DROW dataset, put it into a ./data/DROWv2-data directory (https://github.com/VisualComputingInstitute/DROW/releases), and ran training as it says in the README:

```
python3 bin/train.py --cfg base_drow_drow_cfg.yaml
```

However, the resulting model performance after 40 epochs is considerably lower than what is reported (barely half for each metric reported in the DROW paper).

Is that to be expected since the model in the original paper was trained on 3 classes and this config trains it on people only? Is there any special configuring that needs to be done to the cfgs file to get better performance on just people? I also noticed the DROW3 model checkpoints in the old DROW repo are not compatible with this repo, which was a bummer as I had hoped to sanity check my results against the original checkpoints :sweat_smile: 

Things I tried so far:
1. multiple training attempts to verify what i am seeing is not just noise (results are pretty consistent, the model seems to quickly overfit the data since the training loss dives down below 0.3 within the first 1000 iterations)
2. verifying i am on pytorch 1.6 (I also tried 1.8 and 1.12 and i get similar results)
3. turning on/off augmentations
4. training with `person_only` set to False (that causes severe instability to the point that the training diverges to a negative loss)

Thank you for your time! :smiley:

**Comment by danjia21** [comment]
Hi @john-maidbot 

The code here focuses on single-class (i.e. person) detection. Nothing prevents the method to generate to multi-class, but the training needs to be tuned.

The result reported in the [DROW3](https://arxiv.org/pdf/1804.02463.pdf) and all the subsequent papers are on the test set. During training, the network is evaluated on the validation set, which is considerably harder. That is probably why you are seeing the discrepancies.  

To run a quick test, you can use our pre-trained checkpoint from [here](https://github.com/VisualComputingInstitute/DR-SPAAM-Detector/releases) (drow_e40.pth). I got these numbers
```
python bin/train.py --cfg cfgs/base_drow_drow_cfg.yaml --ckpt <PATH_TO_CKPT> --evaluation

On valiation
AP_0.3 0.434614 peak-F1_0.3 0.495288 EER_0.3 0.493230
AP_0.5 0.450364 peak-F1_0.5 0.502395 EER_0.5 0.500000

On test
AP_0.3 0.637571 peak-F1_0.3 0.643102 EER_0.3 0.640688
AP_0.5 0.658901 peak-F1_0.5 0.650488 EER_0.5 0.648279
```

**Comment by john-maidbot** [comment]
Hi @danjia21 ,
Thank you for the fast reply! 
When I evaluate and test with the checkpoint you mentioned, I get the exact same results as you.
I also reran my model through the evaluation and the test output I get is very similar to yours :+1: I think you are correct that I was getting thrown off by the validation output :sweat_smile:
Thanks again!
